src/System.Text.Json/tests/Serialization/CyclicTests.cs

```diff
@@ -19,6 +19,43 @@ public static void WriteCyclicFail()
             Assert.Throws<JsonException>(() => JsonSerializer.ToString(obj));
         }
 
+        [Theory]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(10)]
+        [InlineData(70)]
+        public static void WriteCyclicFail(int depth)
+        {
+            var rootObj = new TestClassWithCycle("root");
+            CreateObjectHierarchy(1, depth, rootObj);
+
+            {
+                var options = new JsonSerializerOptions();
+                options.MaxDepth = depth + 1;
+
+                // No exception
+                JsonSerializer.ToString(rootObj, options);
```

Comment:
Can we add an assert saying the returning string is not empty?

Reply:
Sure